libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -165,11 +166,16 @@ pub(super) fn merge_local_folder_manifest(
         return MergeLocalFolderManifestOutcome::NoChange;
     }
 
+    let mut local_from_remote = LocalFolderManifest::from_remote_with_local_context(
```

Comment:
This is not related to the PR, but I find this method poorly designed: we are both applying remote sync pattern (which is normal since we are dealing with remote data) AND restoring local sync pattern (which by definition reference files that are in local only...)

I'm wondering how this behave in corner cases (e.g. if a new file in the remote manifest match the local prevent sync pattern, and a file with the same name is also in local)

So my guess here is `LocalFolderManifest::from_remote_with_local_context` shouldn't exist in the first place: it is the job of `merge_local_folder_manifest` to provide a new local folder manifest when given a remote and a local.

Reply:
I've created an issue for that #7738